linkerd/http/prom/src/body_data/metrics.rs

```diff
@@ -0,0 +1,94 @@
+//! Prometheus counters for request and response bodies.
+
+use linkerd_metrics::prom::{self, Counter, Family, Registry};
+
+/// Counters for response body frames.
+#[derive(Clone, Debug)]
+pub struct ResponseBodyFamilies<L> {
+    /// Counts the number of response body frames.
+    resp_body_frames_total: Family<L, Counter>,
+    /// Counts the total number of bytes in response body frames.
+    resp_body_frames_bytes: Family<L, Counter>,
+}
+
+/// Counters to instrument a request or response body.
+#[derive(Clone, Debug, Default)]
+pub struct BodyDataMetrics {
+    /// Counts the number of request body frames.
+    pub frames_total: Counter,
+    /// Counts the total number of bytes in request body frames.
+    pub frames_bytes: Counter,
```

Comment:
unfortunately these need to be public, because our test code lives in `linkerd-app-outbound`. applying this change, the tests are no longer able to compile.

```
/workspaces/linkerd2-proxy$ cargo check --target-dir=target-devcontainer/ --message-format=short --workspace --tests
linkerd/app/outbound/src/http/logical/policy/route/backend/metrics/tests.rs:158:9: error[E0451]: field `frames_total` of struct `BodyDataMetrics` is private
linkerd/app/outbound/src/http/logical/policy/route/backend/metrics/tests.rs:159:9: error[E0451]: field `frames_bytes` of struct `BodyDataMetrics` is private
```

the offending snippet is...

```rust
    // Acquire the counters for this backend.
    tracing::info!("acquiring response body metrics");
    let labels = labels::RouteBackend(parent_ref.clone(), route_ref.clone(), backend_ref.clone());
    let BodyDataMetrics {
        frames_total,
        frames_bytes,
    } = metrics.get_response_body_metrics(&labels);
```

i started on a patch to this via some `#[cfg(test)]` accessors, but realized it's nullified if we're going to be moving to a `Histogram` (_which doesn't provide a way to read its buckets_).

i'll keep this open until i've finished that, but noting why these are `pub` in the meantime.

Reply:
on further inspection, `#[cfg(test)]` accessors are not a viable route, because that flag does not propagate to dependencies when running unit tests. unfortunate, because that's a pattern i quite like!

~~these fields~~ this field (_now a single histogram, via 225674b_) will need to remain public to account for test code.